### PR TITLE
Updates to physchem data and metrics

### PR DESCRIPTION
**Pull request by mlgill**
Contains the following updates:

**Datasets**
- [x] Updates to physchem datasets and fingerprints according to AZ data
- [x] Updates to bioactivity datasets and fingerprints according to AZ data
- [x] Updates to dataset classes used in metrics calculation
- [x] Updates to benchmark script

**Metrics code**
- [x] Review of metrics code

**See also the following notes**:
* Bug fix for novelty metric -- previously was calculating the number of molecules in training data, not the number NOT in training data: https://github.com/rilango/cheminformatics/pull/7#discussion_r754578899
* Number of CV splits is configurable https://github.com/rilango/cheminformatics/pull/7#discussion_r754577708
* Invalid smiles now removed from unique and novelty metrics: https://github.com/rilango/cheminformatics/pull/7#discussion_r754578324

Have moved plotting to separate PR.

### CODE REVIEW

File: benchmark/cuchembm/__main__.py [hunk 1]
```diff
@@ -169,23 +174,29 @@ def main(cfg):
                 estimator, param_dict = metric.model_dict[iter_val]
                 kwargs.update({'estimator': estimator, 'param_dict': param_dict})
                 if metric.name.endswith('bioactivity'):
+                    kwargs['n_splits'] = cfg.metric.modelability.bioactivity.n_splits
```
Comment (mlgill) [inline benchmark/cuchembm/__main__.py:177]: @rilango Made the number of CV splits configurable parameter

File: benchmark/cuchembm/metrics/sampling.py [hunk 2]
```diff
@@ -143,16 +150,20 @@ def sample(self, smiles, num_samples, radius):
                                                       scaled_radius=radius,
                                                       force_unique=False,
                                                       sanitize=False)
+        generated_smiles = generated_smiles[1:]
+        if self.remove_invalid:
```
Comment (mlgill) [inline benchmark/cuchembm/metrics/sampling.py:155]: @rilango Now removing invalid SMILES by default from uniqueness and novelty metrics

File: benchmark/cuchembm/metrics/sampling.py [hunk 3]
```diff
@@ -161,9 +172,9 @@ def variations(self, cfg, **kwargs):
         radius_list = [float(x) for x in radius_list]
         return {'radius': radius_list}
 
-    def smiles_in_train(self, smiles):
+    def smiles_not_in_train(self, smiles):
         in_train = self.training_data.is_known_smiles(smiles)
-        return in_train
+        return not(in_train)
```
Comment (mlgill) [inline benchmark/cuchembm/metrics/sampling.py:179]: @rilango Note: I believe there was a bug in this metric -- should be counting the number of molecules NOT in training data.
